index.js

```diff
@@ -170,10 +170,12 @@ Object.defineProperty(Primus.prototype, 'client', {
 //
 Object.defineProperty(Primus.prototype, 'Socket', {
   get: function () {
-    return require('load').compiler(this.library(true), 'primus.js', {
-      __filename: 'primus.js',
-      __dirname: process.cwd()
-    }).Primus;
+    var compiler = new module.constructor();
```

Comment:
Or we can do something like
```js
var Module = module.constructor;
  , compiler = new Module();
```
Don't know if we need to actually call `require`.